pkg/install/0-installstorage.go

```diff
@@ -83,119 +82,108 @@ func (i *Installer) installStorage(ctx context.Context, installConfig *installco
-										Priority:                 to.Int32Ptr(103),
-										Direction:                mgmtnetwork.SecurityRuleDirectionInbound,
-									},
-									Name: to.StringPtr("bootstrap_ssh_in"),
+								Name: to.StringPtr("apiserver_in"),
+							},
+							{
+								SecurityRulePropertiesFormat: &mgmtnetwork.SecurityRulePropertiesFormat{
+									Protocol:                 mgmtnetwork.SecurityRuleProtocolTCP,
+									SourcePortRange:          to.StringPtr("*"),
+									DestinationPortRange:     to.StringPtr("22"),
+									SourceAddressPrefix:      to.StringPtr("*"),
+									DestinationAddressPrefix: to.StringPtr("*"),
+									Access:                   mgmtnetwork.SecurityRuleAccessAllow,
+									Priority:                 to.Int32Ptr(103),
+									Direction:                mgmtnetwork.SecurityRuleDirectionInbound,
 								},
+								Name: to.StringPtr("bootstrap_ssh_in"),
 							},
 						},
-						Name:     to.StringPtr("aro-controlplane-nsg"),
-						Type:     to.StringPtr("Microsoft.Network/networkSecurityGroups"),
-						Location: &installConfig.Config.Azure.Region,
-					},
-					APIVersion: apiVersions["network"],
-				},
-				{
-					Resource: &mgmtnetwork.SecurityGroup{
-						Name:     to.StringPtr("aro-node-nsg"),
-						Type:     to.StringPtr("Microsoft.Network/networkSecurityGroups"),
-						Location: &installConfig.Config.Azure.Region,
 					},
-					APIVersion: apiVersions["network"],
+					Name:     to.StringPtr("aro-controlplane-nsg"),
+					Type:     to.StringPtr("Microsoft.Network/networkSecurityGroups"),
+					Location: &installConfig.Config.Azure.Region,
 				},
+				APIVersion: apiVersions["network"],
 			},
-		}
-
-		i.log.Print("deploying storage template")
-		err = i.deployments.CreateOrUpdateAndWait(ctx, resourceGroup, "azuredeploy", mgmtresources.Deployment{
-			Properties: &mgmtresources.DeploymentProperties{
-				Template: t,
-				Mode:     mgmtresources.Incremental,
+			{
+				Resource: &mgmtnetwork.SecurityGroup{
+					Name:     to.StringPtr("aro-node-nsg"),
+					Type:     to.StringPtr("Microsoft.Network/networkSecurityGroups"),
+					Location: &installConfig.Config.Azure.Region,
+				},
+				APIVersion: apiVersions["network"],
 			},
-		})
-		if err != nil {
-			if isDeploymentActiveError(err) {
-				i.log.Print("waiting for storage template")
-				err = i.deployments.Wait(ctx, resourceGroup, "azuredeploy")
-			}
-			isQuota, errMsg := isResourceQuotaExceededError(err)
-			if isQuota {
-				return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeQuotaExceeded, errMsg, "")
-			}
-			return err
-		}
+		},
 	}
 
-	{
+	i.log.Print("deploying storage template")
+	err = i.deployments.CreateOrUpdateAndWait(ctx, resourceGroup, "azuredeploy", mgmtresources.Deployment{
+		Properties: &mgmtresources.DeploymentProperties{
+			Template: t,
+			Mode:     mgmtresources.Incremental,
+		},
+	})
+	if err != nil {
+		err = i.checkDeploymentErr(ctx, err, resourceGroup)
```

Comment:
this does reduce code, but I'm not sure it clarifies.  Would it be better to make a new function that does the CreateOrUpdateAndWait + the error handling in one place?

Reply:
Yes, I like that idea.